### ISSUE TITLE
Add CO2 emission to the table

### ISSUE THREAD

**Issue by joshuawe** [pbogre/jetlog#53]
Including the CO2 consumption per flight (and perhaps later as a statistic over months or years) would be a great feature.

There are websites/APIs that calculate the consumption:

- Amadeus has a an easy one, including an example: https://developers.amadeus.com/blog/get-flight-co2-emissions-data-using-amadeus-flight-api
- GoClimate could also be an option: https://www.goclimate.com/blog/flight-emissions-api/

**Comment by pbogre** [comment]
Thanks for this issue. It's not the first time someone expresses interest in this feature.

At first I put it off because of the complexity of estimating CO2 emission. I have looked at the APIs you have linked and they look quite interesting, but if possible I would like to implement the calculation myself,  rather than calling an external API.

One of the APIs you mention has a paper about how their estimation works ([link](https://www.goclimate.com/blog/wp-content/uploads/2019/04/Calculations-in-GoClimateNeutral-Flight-Footprint-API.pdf)), so once I get started on this feature i'll definitely look into that.

Thanks

**Comment by joshuawe** [comment]
It would be ideal, if it could be done without any API calls.
If doing the calculations independently it would be a very rudimentary estimate as airplane type, number of passengers, cargo, etc. influence the calculations and these would needed to be retrieved via some API call.

One solution if robustness to API availability could be to calculate CO2 emissions with an API call but have a fallback estimate for when the API calls fail.

The document that you linked to unfortunately only provides a high-level explanation but no formulas.

**Comment by pbogre** [comment]
You are right that the estimation would be rudimentary, however looking at the request parameters for the APIs you linked shows that they simply take origin and destination airports as parameters (as well as cabin class), and derive an estimate from that. Does this mean that their calculations are also only dependent on the distance flown?

About using an API and local as fallback, if I had to use time to implement a CO2 estimation, I would rather work on make it accurate enough that it can substitute the API calls. Otherwise I would rather use the APIs 100% of the time instead of having some flights where the estimation has a large error.

**Comment by Floo33R** [comment]
I saw recently, that the emissions for Lufthansa flights are displayed in the app with the boarding pass. Maybe it's possible to add the field and enter it manually. Would also add some nice statistics.

**Comment by pbogre** [comment]
@Floo33R i feel like it is possible for the airlines to provide this data because they have much more information about the flight. I did some digging, and their CO2 calculator uses (and is not limited to) "the flight route, the average fuel consumption along this route, the travel class and the seat occupancy rate." (https://www.lufthansa.com/gb/en/fly-more-sustainabl.html#calculation).

Some of this information is simply not available to Jetlog or even to the user, like the seat occupancy. Additionally, the calculation would be very dependent on the aircraft model, and the way it currently works is that there's no official aircraft database in Jetlog.

IMO this would indeed be a very good addition to have, but the only feasible way to do it without wildly incorrect estimates is some sort of API that calculates it for you given the flight number, or just some external service in general.

**Comment by Floo33R** [comment]
I just saw it on me E-Tickets, that the airline provides that information. That's why i thought this could also be an option. 
Will also keep my eyes open.
